llvm/docs/SYCLOffloadingDesign.rst

```diff
@@ -0,0 +1,677 @@
+tested before deploying.
+
+Remainder of this document illustrates the steps necessary to compile a SYCL
+offloading application using clang.
+
+*******************************************
+High-level SYCL Offloading Compilation Flow
+*******************************************
+
+.. figure:: figures/toplevel.png
+  :scale: 50%
+  :align: center
+  
+  Figure 1: Top-level SYCL offloading compilation flow.
+
+Figure 1 showcases the SYCL offloading compilation flow. The SYCL compiler
+performs the following high-level actions to generate SYCL offloading code:
+
+  - Front-end device compilation step (`clang --target=<device triple>`)
+  - Packaging of device code (`clang-offload-packager`)
+  - Front-end host compilation step (`clang --target=<host triple> --fembed-offload-object`)
+  - Linking stage (`clang-linker-wrapper`)
+
+The set of possible 'clang -cc1' invocations for a given user command is shown
+below. 
+
+.. code-block:: console
+  :linenos:
+
+    $ clang++ -fsycl --offload-arch=pvc,sm_52 test.cpp -###
+    clang -cc1 -triple nvptx64 ...     // AOT device compilation for sm_52
+    clang -cc1 -triple spir64_gen ...  // AOT device compilation for pvc
+    clang -cc1 -triple spirv64 ...     // JIT device compilation (backup) for pvc
+    clang -cc1 -triple x86_64 ...      // Host compilation
+
+****************************
+Front-end device compilation
+****************************
+This step parses input source, outlines device part of the code, applies
+additional restrictions on the device code (e.g. no exceptions or virtual
+calls), generates LLVM IR bitcode for the device code only and "integration
+header" which provides information like kernel name, parameters order and data
+type for the runtime library. Multiple LLVM IR bitcodes (in case of multiple
+targets) are packaged into a single object by the clang-offload-packager.
+
+An example of front-end device compilation command is shown below:
+
+  .. code-block:: console
+
+    $ clang -cc1 -triple spirv64 -fsycl-is-device test.cpp -o test.bc
+    
+Front-end device compilation for SYCL offloading can be split into the following
+components - Device code outlining, SYCL kernel function object lowering,
+Generation of device code diagnostics, and Integration header generation. These
+components are explained :ref:`in this document<sycl_frontend_device_compilation>`.
+
+*********************
+Device code packaging
+*********************
+When dealing with multiple device binaries, an additional step is performed to
+package the multiple device binaries before being added to the host object. This
+additional step is performed with the clang-offload-packager taking image inputs
+containing information relating to the target triple, architecture setting and
+offloading kind.
+
+The clang-offload-packager is run during ‘fat object’ generation regardless of
+the number of device binaries being added to the conglomerate fat object. The
```

Comment:
Please avoid the "fat" terminology as it can have negative connotations. There was a suggestion to use "multiple architecture" terminology or similar in the initial libsycl PR.